src/solaris/meson.build

```diff
@@ -0,0 +1,9 @@
+dep_socket = cc.find_library('socket')
+if not cc.has_function('connect', dependencies: dep_socket)
+  error('required function connect missing')
+endif
```

Comment:
We should not check for this. I think it was necessary with autotools to properly detect the presence of the library.